src/common/paddle/index.ts

```diff
@@ -237,17 +239,29 @@ export const insertClaimableItem = async (
     },
   });
 
-  await con.getRepository(ClaimableItem).insert({
-    type: ClaimableItemTypes.Plus,
-    email: customer.email,
-    flags: {
-      cycle: extractSubscriptionCycle(data.items),
-      createdAt: data.startedAt,
-      subscriptionId: data.id,
-      provider: SubscriptionProvider.Paddle,
-      status: UserSubscriptionStatus.Active,
-    },
-  });
+  const currentSubscription = existingEntries.find(
+    (entry) => entry.flags.subscriptionId === data.id,
+  );
+
+  const flags = {
+    cycle: state ? extractSubscriptionCycle(data.items) : null,
+    createdAt: state ? data.startedAt : null,
+    subscriptionId: state ? data.id : null,
+    provider: state ? SubscriptionProvider.Paddle : null,
+    status: state ? UserSubscriptionStatus.Active : null,
```

Comment:
Just a random thought. I think we don't even have to run updates on this entity. This table is more of a record for when user is trying to claim their subscription. Cancelling it shouldn't matter here since the entity has served its purpose already.

Reply:
Technically user could change subscription type in paddle while still not registered, and then it would claim different sub?

Reply:
Makes sense. Then yeah, let's keep the update all throughout.